### PR TITLE
Make disk export more configurable

### PR DESCRIPTION
**Pull request by d2r2**
Make image disk export even more configurable with implementation of 2 extra variables: `$(MAX_WIDTH)` and `$(MAX_HEIGHT)`. New variables allow to create path template like this `$(FILE_FOLDER)/darktable_exported_$(MAX_WIDTH)x$(MAX_HEIGHT)/$(FILE_NAME)`, which in turn make it possible to create multiple `darktable_exported` folders. In my case, with this template I can generate full resolution images and images limited to 2048x2048 which automatically sorted by 2 folders (`darktable_exported_0x0` and `darktable_exported_2048x2048`).

Implementation steps taken:
1) Add extra parameters (int max_width, int max_height) to dt_imageio_module_storage_t (*store) method. All dt_imageio_module_storage_t implementations were updated.
2) Add public `$(MAX_WIDTH)`, `$(MAX_HEIGHT)` variables to expand, once found in export path template.
3) Lua API doesn't change (store function), since I don't want to break existing scripts. Though it is possible to change in the future.
4) Add russian localization for `$(MAX_WIDTH)`, `$(MAX_HEIGHT)` variables description.

### CODE REVIEW

File: src/common/import_session.c [hunk 1]
```diff
@@ -217,6 +217,14 @@ void dt_import_session_set_filename(struct dt_import_session_t *self, const char
 }
 
 
+/*
```
Comment (TurboGit) [inline src/common/import_session.c]: Either remove or put a comments explaining why is this commented out code.

Comment (TurboGit) [reply src/common/import_session.c]: BTW, I'm not sure to understand why this is needed on session?

Comment (d2r2) [reply src/common/import_session.c]: @TurboGit, I agree. I mistakenly left this code commented out, but certainly it should be removed.

Comment (d2r2) [reply src/common/import_session.c]: @TurboGit, resolved.

**Comment by d2r2** [comment]
@TurboGit, taking this opportunity, want to ask: since this change will go to the new major release, do I have to create same change for branch `darktable-2.6.x`? Again, some time ago I propose Lumix LX-100M2 changes for `darktable-2.6.x`. Will they automatically get into the `master` branch, or I should prepare similar change for `master`?

**Comment by TurboGit** [comment]
Thanks. Merged now.

**Comment by TurboGit** [comment]
> same change for branch darktable-2.6.x

If you want this to be in 2.6, why not but i'm not sure a new 2.6 release will be made as there is some many changes in master that merging will be difficult. You can see if this change is easy enough to merge.

**Comment by TurboGit** [comment]
> Again, some time ago I propose Lumix LX-100M2 changes for `darktable-2.6.x`.

AFAIK the white-balance has been added into master and will be added into 2.6.x later if needed.

**Comment by d2r2** [comment]
@TurboGit, thanks for the clarification!

**Comment by houz** [comment]
could you please tell me why you needed these two variables? the width/height settings in the export module were already max values, that describe the bounding box of the exported image.

**Comment by d2r2** [comment]
@houz, as I describe above, using this new variables, you can generate multiple `darktable_exported` folders, where $(MAX_WIDTH) and $(MAX_HEIGHT) will be constant values over export session, thus let you create, for instance, output folders, kind of `darktable_export_0x0` and `darktable_export_2048x2048`. Or if explained differently, $(MAX_WIDTH) and $(MAX_HEIGHT) is a copy of GUI `max size` field values.

**Comment by houz** [comment]
My point is, the max width/height is already in fdata. Why did you need all those api changes?

**Comment by d2r2** [comment]
@houz, it seems I have missed this opportunity (although I remember that fdata have fields named max_width, max height).

To be exactly sure, I just replace line in my code:
```
dt_variables_set_max_width_height(d->vp, max_width, max_height);
```
with:
```
dt_variables_set_max_width_height(d->vp, fdata->max_width, fdata->max_height);
```
... and as I see everything work as it should.

So, should I create new change with reverting (*store) API change?

**Comment by houz** [comment]
If it's not too much of a problem I would welcome that. We are already piling up too many arguments in these internal functions, and having even more that duplicate other data fields sounds like a sure way to confuse people.

**Comment by d2r2** [comment]
@houz, I agree that it is no sense to change API just for nothing. I will create a correction update later today. Thanks for pointing to this!

**Comment by d2r2** [comment]
@houz, last question: why `gboolean high_quality, gboolean upscale` variables in `(*store)` method not tied to fdata structure? Or this is, as you stated, question of not optimal variable allocation?

**Comment by TurboGit** [comment]
Oh! I missed the fact that the value were already available. So houz is right.

**Comment by d2r2** [comment]
@TurboGit, then I have following question: why gboolean high_quality, gboolean upscale variables in (*store) method not tied to fdata structure as well? Or this is, as houz stated, question of not optimal variable allocation?

**Comment by TurboGit** [comment]
Not sure about this. AFAIK this was done long time ago, maybe we can improve this part. But I would suggest to do that in another PR anyway.

**Comment by d2r2** [comment]
@TurboGit, thanks for reply. OK, for now I will return API to the original view in next PR.

**Pull request merged**